clang/test/CXX/expr/expr.unary/expr.delete/p10.cpp

```diff
@@ -23,3 +30,14 @@ struct alignas(Align) B {
 };
 void f(B<__STDCPP_DEFAULT_NEW_ALIGNMENT__> *p) { delete p; }
 void f(B<__STDCPP_DEFAULT_NEW_ALIGNMENT__ * 2> *p) { delete p; } // expected-error {{deleted}}
+
+// Ensure that a deleted destructor is acceptable when the selected overload
+// for operator delete is a destroying delete. See the comments in GH118660.
+struct S {
+  ~S() = delete;
+  void operator delete(S *, std::destroying_delete_t) noexcept {}
+};
+
+void foo(S *s) {
+  delete s; // Was rejected, is intended to be accepted.
+}
```

Comment:
can you add a test with a private destructor?
(I found this unresolved issue https://cplusplus.github.io/CWG/issues/2889.html that is tangentially relevant )

Reply:
Good call, that identified an issue which I've now fixed.